src/dataflow-types/src/client.rs

```diff
@@ -961,28 +1018,199 @@ pub mod process_local {
+                                self.state = GrpcComputedTcpConn::Connected((tx, resp_rx));
+                            }
+                            Err(err) => {
+                                info!(
+                                    "GrpcComputedClient: Connection to {} refused: {}",
+                                    &self.addr, err
+                                );
+                                self.state = GrpcComputedTcpConn::Disconnected;
+                            }
+                        }
+                    }
+                    GrpcComputedTcpConn::Connected(_) => break,
+                    GrpcComputedTcpConn::Backoff(deadline) => {
+                        time::sleep_until(*deadline).await;
+                        self.state = GrpcComputedTcpConn::Disconnected;
+                    }
+                }
+            }
+        }
+    }
+
+    #[async_trait]
+    impl<T> GenericClient<ComputeCommand<T>, ComputeResponse<T>> for GrpcComputedClient
+    where
+        T: ProtoSerializableTimestamp + 'static + std::fmt::Debug + Send,
+    {
+        async fn send(&mut self, cmd: ComputeCommand<T>) -> Result<(), anyhow::Error> {
+            let sender = if let GrpcComputedTcpConn::Connected((sender, _)) = &self.state {
+                sender
+            } else {
+                return Err(anyhow::anyhow!("Sent into disconnected channel"));
+            };
+            sender
+                .send(ProtoSerializableTimestamp::compute_command_into_proto(cmd))
+                .await
+                .expect("Ok");
+            Ok(())
+        }
 
+        async fn recv(&mut self) -> Result<Option<ComputeResponse<T>>, anyhow::Error> {
+            if let GrpcComputedTcpConn::Connected(channels) = &mut self.state {
+                match channels.1.next().await {
+                    Some(Ok(x)) => Ok(Some(
+                        ProtoSerializableTimestamp::compute_response_into_rust(x)?,
+                    )),
+                    other => {
+                        match other {
+                            Some(Ok(_)) => unreachable!("handled above"),
+                            None => error!("connection unexpectedly terminated cleanly"),
+                            Some(Err(e)) => error!("connection unexpectedly errored: {}", e),
+                        }
+
+                        self.state = GrpcComputedTcpConn::Disconnected;
+                        Err(anyhow::anyhow!("Connection severed"))
+                    }
+                }
+            } else {
+                Err(anyhow::anyhow!("Connection severed"))
+            }
+        }
+    }
+
+    #[async_trait]
+    impl Reconnect for GrpcComputedClient {
+        fn disconnect(&mut self) {
+            info!("GrpcComputedClient: disconnect called");
+            self.state = GrpcComputedTcpConn::Disconnected;
+        }
+
+        async fn reconnect(&mut self) {
+            info!("GrpcComputedClient: reconnect called");
```

Comment:
Might want to change `info!` to something less noisy?

Reply:
yeah these should all be `debug!`s!

Reply:
I did all to debug! except, "start listening on X", "connected" and "connection refused". I think these might be useful messages (and not very noisy)